ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/customercenter/navigation/CustomerCenterNavigationState.kt

```diff
@@ -0,0 +1,68 @@
+package com.revenuecat.purchases.ui.revenuecatui.customercenter.navigation
+
+import androidx.compose.runtime.Immutable
+import java.util.ArrayDeque
+import java.util.Deque
+
+@Immutable
+internal data class CustomerCenterNavigationState(
```

Comment:
Not for this PR! I mentioned it over Slack but I wonder if we might want to make this more generic so it can be reused by paywalls or any other UI in the future... For example, I think it might be cool to share any logic for displaying a sheet here, which we recently added in Paywalls... Having said all that, I'm ok making it focus on Customer center for now, since that's the only thing it's going to use it. We can always revisit this point in the future.

Reply:
I agree. I will focus on the Customer Center for now and revisit it because I am worried I would have to redo a lot of things. Making sure I am not breaking any flow is a bit cumbersome and I don't want to delay the project more